src/lib/core/WeaveMessageLayer.cpp

```diff
@@ -2488,5 +2489,32 @@ void WeaveMessageLayer::GetPeerDescription(char * buf, size_t bufSize, const Wea
         msgInfo->InCon);
 }
 
+/**
+ * @brief
+ *   Generate random Weave device Id.
+ *
+ * @details
+ *   This function generates 64-bit locally unique Weave device Id. This function uses cryptographically strong
+ *   random data source to guarantee uniqueness of generated value. Note that bit 56 of the generated Weave
+ *   device Id is set to 1 to indicate that generated Weave device Id is locally (not globally) unique.
+ *
+ * @param deviceId                      A reference to the 64-bit Weave device Id.
+ *
+ * @retval  #WEAVE_NO_ERROR             If Weave device Id was successfully generated.
+ */
+NL_DLL_EXPORT WEAVE_ERROR GenerateWeaveDeviceId(uint64_t & deviceId)
+{
+    WEAVE_ERROR err;
+    enum { kMask_WeaveDeviceIdLocallyUnique = 0x0100000000000000ULL };
```

Comment:
The bit value used here is wrong.  It should be 0x0200000000000000ULL.

Rather than defining a enum locally, I suggest we generalize this by defining the following enum in WeaveDeviceLayer.h:

```
/**
 * Bit field definitions for IEEE EUI-64 Identifiers.
 */
enum
{
    kEUI64_UL_Mask          = 0x0200000000000000ULL,        /**< Bitmask for the Universal/Local (U/L) bit within an EUI-64 identifier.
                                                                 A value of 0 indicates the id is Universally (globally) administered.
                                                                 A value of 1 indicates the id is Locally administered. */

    kEUI64_UL_Unversal      = 0,                            /**< Universal/Local bit value indicating a Universally administered EUI-64 identifier. */

    kEUI64_UL_Local         = kEUI64_UL_Mask,               /**< Universal/Local bit value indicating a Locally administered EUI-64 identifier. */

    kEUI64_IG_Mask          = 0x0100000000000000ULL,        /**< Bitmask for the Individual/Group (I/G) bit within an EUI-64 identifier.
                                                                 A value of 0 indicates the id is an individual address.
                                                                 A value of 1 indicates the id is a group address. */

    kEUI64_IG_Individual    = 0,                            /**< Individual/Group bit value indicating an individual address EUI-64 identifier. */

    kEUI64_IG_Group         = kEUI64_IG_Mask,               /**< Individual/Group bit value indicating an group address EUI-64 identifier. */
};
```

Reply:
done